packages/auth/src/accessToken/routes.test.ts

```diff
@@ -0,0 +1,210 @@
+    }
+  )
+
+  const BASE_GRANT = {
+    state: GrantState.Pending,
+    startMethod: [StartMethod.Redirect],
+    continueToken: crypto.randomBytes(8).toString('hex').toUpperCase(),
+    continueId: v4(),
+    finishMethod: FinishMethod.Redirect,
+    finishUri: 'https://example.com/finish',
+    clientNonce: crypto.randomBytes(8).toString('hex').toUpperCase(),
+    clientKeyId: KEY_REGISTRY_ORIGIN + TEST_KID_PATH,
+    interactId: v4(),
+    interactRef: crypto.randomBytes(8).toString('hex').toUpperCase(),
+    interactNonce: crypto.randomBytes(8).toString('hex').toUpperCase()
+  }
+
+  const BASE_ACCESS = {
+    type: AccessType.OutgoingPayment,
+    actions: [Action.Read, Action.Create],
+    limits: {
+      receivingAccount: 'https://wallet.com/alice',
+      sendAmount: {
+        value: '400',
+        assetCode: 'USD',
+        assetScale: 2
+      }
+    }
+  }
+
+  const BASE_TOKEN = {
+    value: crypto.randomBytes(8).toString('hex').toUpperCase(),
+    managementId: 'https://example.com/manage/12345',
+    expiresIn: 3600
+  }
+
+  describe('Introspect', (): void => {
+    let grant: Grant
+    let access: Access
+    let token: AccessToken
+    beforeEach(
+      async (): Promise<void> => {
+        grant = await Grant.query(trx).insertAndFetch({
+          ...BASE_GRANT
+        })
+        access = await Access.query(trx).insertAndFetch({
+          grantId: grant.id,
+          ...BASE_ACCESS
+        })
+        token = await AccessToken.query(trx).insertAndFetch({
+          grantId: grant.id,
+          ...BASE_TOKEN
+        })
+      }
+    )
+    test('Cannot introspect fake token', async (): Promise<void> => {
+      const ctx = createContext(
+        {
+          headers: { Accept: 'application/json' }
+        },
+        {}
+      )
+      ctx.request.body = {
+        access_token: v4(),
+        proof: 'httpsig',
+        resource_server: 'test'
+      }
+      await expect(accessTokenRoutes.introspect(ctx)).rejects.toMatchObject({
+        status: 404,
+        message: 'token not found'
+      })
+    })
+
+    test('Cannot introspect if no token passed', async (): Promise<void> => {
+      const ctx = createContext(
+        {
+          headers: { Accept: 'application/json' }
+        },
+        {}
+      )
+      ctx.request.body = {
+        proof: 'httpsig',
+        resource_server: 'test'
+      }
+      await expect(accessTokenRoutes.introspect(ctx)).rejects.toMatchObject({
+        status: 400,
+        message: 'invalid introspection request'
+      })
+    })
+
+    test('Successfully introspects valid token', async (): Promise<void> => {
+      const scope = nock(KEY_REGISTRY_ORIGIN)
+        .get(TEST_KID_PATH)
+        .reply(200, {
+          keys: [TEST_CLIENT_KEY]
+        })
+      const ctx = createContext(
+        {
+          headers: { Accept: 'application/json' }
+        },
+        {}
+      )
+      ctx.request.body = {
+        access_token: token.value,
+        proof: 'httpsig',
```

Comment:
Just as a general question, are we using the GET request to `openpayments.network` as our method of determining client key possession instead of the `httpsig` method?